### ISSUE TITLE
backups in not creating

### ISSUE THREAD

**Issue by dampelz** [Froxlor/Froxlor#1038]
**Describe the bug**
Backups is not creating, i can't find any backup archives or files.
Only this: /backup/.tmp/mysql
all folders is emply

My task

![Screenshot 2022-06-27 at 14-00-03 anton - Froxlor Server Management Panel](https://user-images.githubusercontent.com/59340295/175926727-d13002af-fbd8-45e8-be18-81774b30d801.png)

**System information**
* Froxlor version: 0.10.35.1-1 (DB: 202112310)
* Web server: nginx
* DNS server: Bind
* POP/IMAP server: Courier/Dovecot
* SMTP server: postfix/exim
* FTP server: proftpd
* OS/Version: Ubuntu 20.04

**To Reproduce**
1. Create the task with folder /backup
2. Waiting when cron task is execute
3. Check the backup folder /var/customers/webs/USER/backup
4. backup folder is empty , only .tmp and mysql folder

backups is enabled for customers

**Comment by d00p** [comment]
well, if there's still a .tmp file, maybe the cron i still running? it's not like a few GB are packed within a second

**Comment by dampelz** [comment]
Where i can find backup logs ?

**Comment by dampelz** [comment]
running via ssh

`/usr/bin/nice -n 5 /usr/bin/php -q /var/www/html/froxlor/scripts/froxlor_master_cronjob.php --backup`

```
SECURITY CHECK FAILED!
The execute string 'mysqldump -u 'root' -p1@ELCFOP1Fa~}|rEQ7vmLrhmu6G2J726viZRgmCEEWRT3u|~JC antonsql1 > /var/customers/webs/anton/backup/.tmp/mysql/antonsql1_202206280134.sql' is a possible security risk!
Please check your whole server for security problems by hand!
```

**Comment by d00p** [comment]
Okay, seems to be the pipe in the password. We may need.to check the escaping there

**Comment by dampelz** [comment]
Ubuntu 20, mysql 8
it's related with new function **password validation** for mysql 8
The password must contain special characters

Please. check it.

**Comment by dampelz** [comment]
Working, but sql files is empty.

`/usr/bin/nice -n 5 /usr/bin/php -q /var/www/html/froxlor/scripts/froxlor_master_cronjob.php --backup`

```
root@tinycp:~# mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxnL84Vb'.
mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxnL84Vb'.
mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxIjQrba'.
```

**Comment by dampelz** [comment]
What should i do

**Comment by d00p** [comment]
Could you comment out the `unlink($mysqlcnf_file)` part and see if the content of the file is what it's supposed to be:

```
[mysqldump]
password=your-password
```

**Comment by dampelz** [comment]
> Could you comment out the unlink($mysqlcnf_file)
Not helping

```
root@tinycp:~# mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxnL84Vb'.
mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxnL84Vb'.
mysqldump: [ERROR] unknown variable 'defaults-file=/tmp/frxIjQrba'.
```
standart mysqldump is working

`root@tinycp:~# mysqldump -v -h127.0.0.1 -uroot -p antonsql1 > /tmp/antonsql1.sql`

**Comment by dampelz** [comment]
```
and see if the content of the file is what it's supposed to be:

[mysqldump]
password=your-password

```

in this file? /etc/mysql/conf.d/mysqldump.cnf ?

**Comment by d00p** [comment]
No, the temporary file created by the cron (hence the commenting out of `unlink` so it does not get deleted afterwards).
The filename is the one passed to the `--defaults-file=` parameter from your error messages

**Comment by dampelz** [comment]
```
[mysqldump]
password=1@ELCFOP1Fa~}|rEQ7vmLrhmu6G2J726viZRgmCEEWRT3u|~JC
```

**Comment by d00p** [comment]
So the defaults-file is created correctly. I just read that the defaults-file parameter needs to be the first, so that might be the root-issue here :) The correct line would now be:

```diff
-                               FileDir::safe_exec('mysqldump -u ' . escapeshellarg($sql_root['user']) . ' -p' . $sql_root['passwd'] . ' ' . $row['databasename'] . ' > ' . FileDir::makeCorrectFile($tmpdir . '/mysql/' . $row['databasename'] . '_' . date('YmdHi', time()) . '.sql'), $bool_false, [
+                               FileDir::safe_exec('mysqldump --defaults-file=' . escapeshellarg($mysqlcnf_file) .' -u ' . escapeshellarg($sql_root['user']) . ' ' . $row['databasename'] . ' > ' . FileDir::makeCorrectFile($tmpdir . '/mysql/' . $row['databasename'] . '_' . date('YmdHi', time()) . '.sql'), $bool_false, [
```

**Comment by d00p** [comment]
Yes will be part of the next release

**Comment by dampelz** [comment]
Ok, you can close this issue.
Thank you.

**Comment by d00p** [comment]
will automatically be closed on commit :)
